openmc/data/product.py

```diff
@@ -49,6 +49,20 @@ def __init__(self, particle='neutron'):
         self.applicability = []
         self.yield_ = 1
 
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        else:
+            eqval = True
+            if (self.particle != other.particle or
+                self.decay_rate != other.decay_rate or
+                self.emission_mode != other.emission_mode or
+                self.distribution.all() != other.distribution.all() or
+                self.applicability.all() != other.applicability.all() or
```

Comment:
`self.distribution` and `self.applicability` are lists so they don't have `all()` methods